www/js/app.js

```diff
@@ -569,19 +602,38 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
      * @param {String} value The chosen content injection mode : 'jquery' or 'serviceworker'
      */
     function setContentInjectionMode(value) {
+        contentInjectionMode = value;
         if (value === 'jquery') {
-            if (isServiceWorkerReady()) {
-                // We need to disable the ServiceWorker
-                // Unregistering it does not seem to work as expected : the ServiceWorker
-                // is indeed unregistered but still active...
-                // So we have to disable it manually (even if it's still registered and active)
-                navigator.serviceWorker.controller.postMessage({'action': 'disable'});
-                messageChannel = null;
+            if (params.extensionURL) {
+                // We are in an extension, and the user may wish to revert to local code
+                var message = 'This will switch to using locally packaged code only. Configuration settings may be lost.\n\n' +
+                'WARNING: App will re-load in JQuery mode!';
+                var launchLocal = function () {
+                    settingsStore.setItem('allowInternetAccess', false, Infinity);
```

Comment:
Storing allowInternetAccess is being used as a passthrough. It is more important in the local extension, where it will allow the app to go straight through to the Service Worker mode without checking first with the user. It is set here just to keep the logic synchronized.